File: code-samples/webhooks/webhook-server.rb
require 'sinatra'
set :port, 5000
post '/webhook' do
    status 200
    headers('Content-Type' => 'application/json')
    headers('Validation-Token' => request.env['HTTP_VALIDATION_TOKEN'])
    if request.env['HTTP_VALIDATION_TOKEN']
      request.body.rewind
    end
    body = request.body.read
    puts body
    # do whatever with body
    body 'OK'
end

File: code-samples/webhooks/webhook-server.js
var http = require('http');
PORT  = 3000

var server = http.createServer(function(req, res) {
  if (req.method == 'POST') {
    if (req.url == "/webhook") {
      if (req.headers.hasOwnProperty("validation-token")) {
        res.setHeader('Content-type', 'application/json');
        res.setHeader('Validation-Token', req.headers['validation-token']);
        res.statusCode = 200;
        res.end();
      } else {
        var body = []
        req.on('data', function(chunk) {
          body.push(chunk);
        }).on('end', function() {
          body = Buffer.concat(body).toString();
          var jsonObj = JSON.parse(body)
          console.log(jsonObj.body);
        });
      }
    }
  }
});

server.listen(PORT);
console.log(`Listening on port ${PORT}`)
